src/App.js

```diff
@@ -2,15 +2,65 @@ import React from 'react';
 import './App.scss';
 
 class App extends React.Component {
-  addOne = () => {};
+  addOne = () => {
+    this.setState(prevState => ({
+      value: prevState.value + 1,
+    }));
+  };
 
-  add100 = () => {};
+  add100 = () => {
+    this.setState(prevState => ({
+      value: prevState.value + 100,
+    }));
+  };
 
-  increase = () => {};
+  increase = () => {
+    if (this.state.value % 5 === 0) {
+      this.add100();
+    }
+
+    this.addOne();
+  };
+
+  render() {
+    return (
+      <Counter />
+    );
+  }
+}
+
+class Counter extends App {
```

Comment:
```suggestion
class Counter extends React.Component {
```
Let's follow the component approach in react when all components are independent. Make separate components for each logical unit in a separate file.
And use state and all methods only for the `Counter`